### PR TITLE
Allow redirects from and for subclasses of page and document

### PR DESCRIPTION
**Pull request by RadoslavK**
Implements the issue #203

### CODE REVIEW

File: lib/jekyll-redirect-from/generator.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def generate(site)
 
       # Must duplicate pages to modify while in loop
       (site.docs_to_write + site.pages.dup).each do |doc|
-        next unless JekyllRedirectFrom::CLASSES.include?(doc.class)
+        next unless JekyllRedirectFrom::CLASSES.any? { |klass| doc.is_a? klass }
```
Comment (parkr) [inline lib/jekyll-redirect-from/generator.rb]: Let's put this into its own method.

```ruby
def redirectable_document?(doc)
  doc.is_a?(Jekyll::Document) || doc.is_a?(Jekyll::Page)
end
```

Then do `next unless redirectable_document?(doc)`

**Comment by parkr** [comment]
Thank you so much! The code looks great. Let's get a quick test in here.

In `spec/jekyll_redirect_from/generator_spec.rb`, let's add:

```ruby
context "redirectable_document?" do
  let(:generator) { JekyllRedirectFrom::Generator.new }
  it "accepts subclasses of Jekyll::Document" do
    SubclassOfJekyllDocument = Class.new(Jekyll::Document)
    expect(generator.send(:redirectable_document?, SubclassOfJekyllDocument.new)).to be_truthy
  end

  it "accepts subclasses of Jekyll::Document" do
    SubclassOfJekyllPage = Class.new(Jekyll::Page)
    expect(generator.send(:redirectable_document?, SubclassOfJekyllPage.new)).to be_truthy
  end
end
```

I haven't run the above code but hopefully it gets you started. 🚀

**Comment by RadoslavK** [comment]
Sorry guys, busy life happened. I have added the tests with a little modification.

**Comment by DirtyF** [comment]
@jekyllbot: merge +minor

**Pull request merged**